vms/evm/uptimetracker/state.go

```diff
@@ -176,33 +157,31 @@ func (s *state) deleteValidator(validationID ids.ID) bool {
 	delete(s.validators, v.validationID)
 	delete(s.nodeIDsToValidationIDs, v.NodeID)
 
-	s.updates[v.validationID] = deletedStatus
+	s.deletedValidators.Add(v.validationID)
 
 	return true
 }
 
 func (s *state) writeState() error {
 	batch := s.db.NewBatch()
 
-	for validationID, updateStatus := range s.updates {
-		switch updateStatus {
-		case updatedStatus:
-			v := s.validators[validationID]
-
-			validatorBytes, err := codecManager.Marshal(codecVersion, v)
-			if err != nil {
-				return fmt.Errorf("failed to marshal validator: %w", err)
-			}
-
-			if err := batch.Put(validationID[:], validatorBytes); err != nil {
-				return fmt.Errorf("failed to put validator: %w", err)
-			}
-		case deletedStatus:
-			if err := batch.Delete(validationID[:]); err != nil {
-				return fmt.Errorf("failed to delete validator: %w", err)
-			}
-		default:
-			return fmt.Errorf("unknown update status: %v", updateStatus)
```

Comment:
We can get rid of this impossible case since we know we only have two cases (updates/deletes)